src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -928,8 +934,9 @@ public void RemoveBlocks(ChainedHeader fork)
 
                 this.UpdateLastBlockSyncedHeight(fork);
 
-                // Reload the addresses and inputs lookup dictionaries.
+                // Reload the lookup dictionaries.
                 this.RefreshInputKeysLookupLock();
+                this.RefreshTxLookupLocked();
```

Comment:
This is a  bad idea, all index data should go in one lock or we will get race conditions.

Reply:
These are within `lockObject` anyway right?

Reply:
ah right sorry my mistake yes correct.